File: ATM Transaction System/ATM_Transaction_System.py
balance = 1000
pin="1234"
def display_menu():
    print("=====ATM Menu=====")
    print("1. View Balance")
    print("2. Deposit Money")
    print("3. Withdraw Money")
    print("4. Change PIN")
    print("5. Exit")
for attempts in range(3):
    entered_pin=input("Enter your PIN: ")
    if entered_pin == pin:
        print("PIN Verified.")
        break
    else:
        print("Invalid PIN!")
else:
    print("Too many wrong attempts. Card Blocked!")
    exit()
while True:
    display_menu()
    choice=input("Enter your choice: ")
    if choice == '1':
        print(f"Your current balance is Rs. {balance}/-")
    elif choice == '2':
        try:
            amount=float(input("Enter the amount to deposit Rs. "))
            if amount > 0:
                balance += amount
                print(f"Rs. {amount}/- is deposited successfully.")
                print(f"Your current balance is Rs. {balance}/-")
            else:
                print("Invalid amount!")
        except ValueError:
            print("Invalid input! Please enter only integers.")
    elif choice == '3':
        try:
            amount=float(input("Enter the amount to withdraw Rs. "))
            if amount > 0 and amount <= balance:
                balance -= amount
                print(f"Rs. {amount} withdrawed successfully.")
                print(f"Your current balance is Rs. {balance}/-")
            else:
                print("Insufficient balance or Invalid amount.")
        except ValueError:
            print("Invalid input! Please enter only integers.")
    elif choice == '4':
        changePIN=input("Enter current PIN: ")
        if changePIN == pin:
            new_PIN=input("Enter new PIN: ")
            confirm_newPIN=input("Re-enter new PIN to confirm: ")
            if new_PIN == confirm_newPIN and new_PIN != "":
                print("PIN changed succcessfully.")
            else:
                print("PINs does not match or empty.")
        else:
            print("Invalid current PIN.")
    elif choice == '5':
        print("Thank You for using ATM. Goodbye!")
        break
    else:
        print("Invalid choice! Please enter 1-5.")

File: ATM Transaction System/ATM_Transaction_System.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
float balance=1000.0;
char pin[5]="1234";
int checkPIN()
{
    char input[5];
    printf("Enter your pin: ");
    scanf("%s",input);
    return strcmp(input, pin) == 0;
}
void viewBalance()
{
    printf("Your current balance is Rs. %.2f/-\n",balance);
}
void depositMoney()
{
    float amount;
    printf("Enter amount to deposit: ");
    scanf("%f",&amount);
    if (amount > 0)
    {
        balance += amount;
        printf("Rs. %.2f/- Deposited successfully.",amount);
    }
    else
    {
        printf("Invalid amount.\n");
    }
}
void withdrawMoney()
{
    float amount;
    printf("Enter amount to withdraw: Rs. ");
    scanf("%f",&amount);
    if (amount <= balance && amount > 0)
    {
        balance -= amount;
        printf("Rs. %.2f/- Withdrawed successfully.\n",amount);
    }
    else
    {
        printf("Insufficient balance or Invalid amount.\n");
    }
}
void changePIN()
{
    char oldPIN[5], newPIN[5];
    printf("Enter current pin: ");
    scanf("%s",oldPIN);
    if (strcmp(oldPIN, pin) == 0)
    {
        printf("Enter new pin: ");
        scanf("%s",newPIN);
        strcpy(pin, newPIN);
        printf("PIN changed successfully.\n");
    }
    else
    {
        printf("Wrong current PIN!\n");
    }
}
int main()
{
    int choice;
    if (!checkPIN())
    {
        printf("Incorrect PIN. Access Denied.\n");
        return 0;
    }
    do 
    {
        printf("\n=====ATM Menu=====\n");
        printf("1. View Balance\n");
        printf("2. Deposit Money\n");
        printf("3. Withdraw Money\n");
        printf("4. Change PIN\n");
        printf("5. Exit\n");
        printf("Enter your choice: ");
        scanf("%d",&choice);
        switch(choice)
        {
            case 1:
                viewBalance();
                break;
            case 2:
                depositMoney();
                break;
            case 3:
                withdrawMoney();
                break;
            case 4:
                changePIN();
                break;
            case 5:
                printf("Thanks for using the ATM. Goodbye!\n");
                break;
            default:
                printf("Invalid option! Try again.\n");
        }
    }while (choice != 5);
    return 0;
}